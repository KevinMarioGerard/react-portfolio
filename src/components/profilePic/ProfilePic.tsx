import * as React from 'react';

export type ProfilePicProps = {
    /**
     * Image source for profile pic
     */
    profilePicSrc: string;
};

const ProfilePic: React.FC<ProfilePicProps> = ({ profilePicSrc }) => {
    return (
        <div className="profile__container">
            <img className="profile__pic" src={profilePicSrc} alt="Profile Pic" />
        </div>
    );
};

export default ProfilePic;
