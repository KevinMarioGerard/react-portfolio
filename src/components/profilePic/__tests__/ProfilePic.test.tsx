import { render } from '@testing-library/react';
import ProfilePic from '../ProfilePic';
import * as React from 'react';

describe('Profile Pic Component tests', () => {

    it('should render without errors', () => {
        expect(() => {
            render(<ProfilePic profilePicSrc="test-image" />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(<ProfilePic profilePicSrc="test-image" />);

        expect(container).toMatchSnapshot();
    });
});