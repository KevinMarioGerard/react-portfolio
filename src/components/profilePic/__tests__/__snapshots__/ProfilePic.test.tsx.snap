// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Pic Component tests should match snapshot 1`] = `
<div>
  <div
    class="profile__container"
  >
    <img
      alt="Profile Pic"
      class="profile__pic"
      src="test-image"
    />
  </div>
</div>
`;
